code/modules/mining/ores_coins.dm

```diff
@@ -523,4 +523,44 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 /obj/item/coin/adamantine/doubloon
 	name = "doubloon"
 
+/obj/item/coin/eldritch
+	name = "eldritch coin"
+	desc = "Everytime it lands heretic you heal, everytime it lands blade you will get damaged. Favours heretics."
+	icon_state = "coin_heretic"
+	custom_materials = list(/datum/material/diamond = 1000, /datum/material/plasma = 1000)
+	sideslist = list("heretic", "blade")
+	material_flags = NONE
+	/// Amount of damage that will be recieved
+	var/coin_damage = 0
+
+/obj/item/coin/eldritch/attack_self(mob/user)
+	if(cooldown < world.time)
+		if(string_attached)
+			to_chat(user, span_warning("The coin won't flip very well with something attached!") )
+			return FALSE
+		cooldown = world.time + 15
+		flick("coin_[coinflip]_flip", src)
+		coinflip = pick(sideslist)
+		icon_state = "coin_[coinflip]"
+		playsound(user.loc, 'sound/items/coinflip.ogg', 50, TRUE)
+		var/oldloc = loc
+		sleep(1.5 SECONDS)
```

Comment:
Does this need to sleep or could we use a timer?

Reply:
I guess this can wait for the coin refactor then